scripts/issue/45/process_bdigg_sushi.py

```diff
@@ -0,0 +1,31 @@
+from brownie import interface
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import registry
+
+
+SLP_TO_WITHDRAW = 5 # ether
+
+def main():
+    """
+    withdraw slp from weth/bdigg position to digg
+    """
+    
+    safe = GreatApeSafe(registry.eth.badger_wallets.treasury_multisig)
+    safe.init_sushi()
+    
+    slp = interface.IUniswapV2Pair(registry.eth.treasury_tokens.slpEthBDigg, owner=safe.account)
+    bdigg = interface.ISettV4h(registry.eth.treasury_tokens.bDIGG, owner=safe.account)
+    digg = interface.ERC20(registry.eth.treasury_tokens.DIGG)
+    weth = interface.ERC20(registry.eth.treasury_tokens.WETH, owner=safe.account)
+    wbtc = registry.eth.treasury_tokens.WBTC
+
+    safe.take_snapshot(tokens=[slp, bdigg, digg])
+    
+    bdigg_before = bdigg.balanceOf(safe)
+    weth_before = weth.balanceOf(safe)
+    safe.sushi.remove_liquidity(False, slp, SLP_TO_WITHDRAW * 10** slp.decimals())
+    bdigg.withdraw(bdigg.balanceOf(safe) - bdigg_before)
+    safe.sushi.swap_tokens_for_tokens(weth, weth.balanceOf(safe) - weth_before, [weth, wbtc, digg], safe.account)
```

Comment:
we are not looking to buy digg. (also, in this case, i think we couldve gone for withdraw_to_one_coin no?). anyway just leave the weth as weth, thats fine.

and just to note: careful here with 'stringing' balanceOf funcs. the weth we get onfork from the remove_liquidity might be different from the amount of weth we will get onchain (due to pool composition changing in between). the script however will then still call the swap with the weth amount hardcoded by the simulation onfork---which could result in a balance too low.

the only time you can call balanceOf correctly in these scripts is in the first action, after that it will not be precise...

Reply:
good to know about how `balanceOf` behaves in that case. Gathering all the bdigg and using `withdrawAll` at the end definitely makes more sense. For `withdraw_to_one_coin`, I believe that is for curve liquidity unless I am missing something

Reply:
sorry, you are right, sushi does not support one sided withdrawal.